src/main/resources/interface-specification.yml

```
@@ -451,6 +451,114 @@ paths:
+                  type: integer
+                description: Remaining requests within time interval
+              'X-Rate-Limit-Interval':
+                schema:
+                  type: integer
+                description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+          '400':
+            description: Bad Request
+            content:
+              application/problem+json:
+                schema:
+                  $ref: '#/components/schemas/Problem'
+            headers:
+              'X-Rate-Limit-Limit':
+                schema:
+                  type: integer
+                description: Max allowed requests within time interval
+              'X-Rate-Limit-Remaining':
+                schema:
+                  type: integer
+                description: Remaining requests within time interval
+              'X-Rate-Limit-Interval':
+                schema:
+                  type: integer
+                description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+          '404':
+            description: Producers not found
+            content:
+              application/problem+json:
+                schema:
+                  $ref: '#/components/schemas/Problem'
+            headers:
+              'X-Rate-Limit-Limit':
+                schema:
+                  type: integer
+                description: Max allowed requests within time interval
+              'X-Rate-Limit-Remaining':
+                schema:
+                  type: integer
+                description: Remaining requests within time interval
+              'X-Rate-Limit-Interval':
+                schema:
+                  type: integer
+                description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+          '429':
+            description: Too Many Requests
+            content:
+              application/problem+json:
+                schema:
+                  $ref: '#/components/schemas/Problem'
+            headers:
+              'X-Rate-Limit-Limit':
+                schema:
+                  type: integer
+                description: Max allowed requests within time interval
+              'X-Rate-Limit-Remaining':
+                schema:
+                  type: integer
+                description: Remaining requests within time interval
+              'X-Rate-Limit-Interval':
+                schema:
+                  type: integer
+                description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available        
```

Rate limiting is applied only in BFF, auth server and api gateway
```suggestion
          '400':
            description: Bad Request
            content:
              application/problem+json:
                schema:
                  $ref: '#/components/schemas/Problem'
          '404':
            description: Producers not found
            content:
              application/problem+json:
                schema:
                  $ref: '#/components/schemas/Problem'
```